src/test/regress/sql/worker_split_copy_test.sql

```diff
@@ -110,8 +110,61 @@ SELECT COUNT(*) FROM worker_split_copy_test."test !/ \n _""dist_123_table_810700
 SELECT COUNT(*) FROM worker_split_copy_test."test !/ \n _""dist_123_table_81070016";
 -- END: List updated row count for local targets shard.
 
+-- Check that GENERATED columns are  handled properly in a shard split operation.
+\c - - - :master_port
+SET search_path TO worker_split_copy_test;
+SET citus.shard_count TO 2;
+SET citus.shard_replication_factor TO 1;
+SET citus.next_shard_id TO 81080000;
+
+-- BEGIN: Create distributed table and insert data.
+CREATE TABLE worker_split_copy_test.dist_table_with_generated_col(id int primary key, new_id int GENERATED ALWAYS AS ( id + 3 ) stored, value char);
+SELECT create_distributed_table('dist_table_with_generated_col', 'id');
+INSERT INTO dist_table_with_generated_col (id, value) (SELECT g.id, 'N' FROM generate_series(1, 1000) AS g(id));
```

Comment:
Would be good to drop the value column here too. To also test for dropped columns, just like is done for shard moves.